src/llmq/context.cpp

```diff
@@ -53,44 +46,33 @@ LLMQContext::LLMQContext(ChainstateManager& chainman, CDeterministicMNManager& d
 
 LLMQContext::~LLMQContext() {
     bls_worker->Stop();
-
-    // LLMQContext doesn't own these objects, but still need to care of them for consistency:
-    llmq::quorumInstantSendManager.reset();
 }
 
 void LLMQContext::Interrupt() {
-    sigman->InterruptWorkerThread();
     shareman->InterruptWorkerThread();
-
-    assert(isman == llmq::quorumInstantSendManager.get());
-    llmq::quorumInstantSendManager->InterruptWorkerThread();
+    sigman->InterruptWorkerThread();
```

Comment:
`sigman` thread starts before `isman`. It means to be stopped in "reversed" order. Firstly, `isman`, after that - `sigman`. Same for `shareman` - if the thread started the first (before sigman and isman) - it means to be stopped last.

This place requires either clarification why order is inversed, or fixing inerruption order.

```cpp
isman->InterruptWorkerThread();
sigman->InterruptWorkerThread();
shareman->InterruptWorkerThread();
```

Reply:
Order corrected and split off into separate commit, resolved in latest push!